### PR TITLE
Fix the flaky e2e test for pending workloads

### LINKED ISSUE #1482: Flaky test e2e test for pending workloads
This test is flaky: "Kueue visibility server when There are pending workloads due to capacity maxed by the admitted job Should allow fetching information about position of pending workloads in ClusterQueue"

Example failure: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/1397/pull-kueue-test-e2e-main-1-27/1736767429989634048

### PR DESCRIPTION
**Pull request by mimowo**
#### What type of PR is this?

/kind bug
/kind flake

#### Which issue(s) this PR fixes:

Fixes #1482

#### Special notes for your reviewer:

Alternative fix is to set very high priority for the blocking job, but I think it does not capture the intention that well.

I reproduced the failure by looping the test locally, it failed 2/1000. After the proposed fix it passed 1000 repeats.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/util/testingjobs/job/wrappers.go [hunk 1]
```diff
@@ -76,7 +76,7 @@ func (j *JobWrapper) BackoffLimit(limit int32) *JobWrapper {
 	return j
 }
 
-func (j *JobWrapper) TerimnationGracePeriod(seconds int64) *JobWrapper {
+func (j *JobWrapper) TerminationGracePeriod(seconds int64) *JobWrapper {
```
Comment (tenzen-y) [inline pkg/util/testingjobs/job/wrappers.go:79]: Oh, this is a great catch :)

File: test/e2e/singlecluster/e2e_test.go [hunk 2]
```diff
@@ -122,12 +122,15 @@ var _ = ginkgo.Describe("Kueue visibility server", func() {
...
-					TerimnationGracePeriod(1).
+					TerminationGracePeriod(1).
 					BackoffLimit(0).
 					PriorityClass(highPriorityClass.Name).
 					Obj()
 				gomega.Expect(k8sClient.Create(ctx, blockingJob)).Should(gomega.Succeed())
 			})
+			ginkgo.By("Ensure the workload is admitted, by awaiting until the job is unsuspended", func() {
+				expectJobUnsuspended(client.ObjectKeyFromObject(blockingJob))
+			})
```
Comment (tenzen-y) [inline test/e2e/singlecluster/e2e_test.go:133]: Should we use `ExpectWorkloadsToBeAdmitted()`?

https://github.com/kubernetes-sigs/kueue/blob/5fced9e6f0b1323153d2df65816e70a087c8f6f7/test/util/util.go#L268

**Review by tenzen-y — COMMENTED** [review]
Otherwise lgtm

File: test/e2e/singlecluster/e2e_test.go (hunk 2 above)
Comment (mimowo) [reply test/e2e/singlecluster/e2e_test.go:133]: In e2e tests we tend to use the API that is used by users, whereas Workload is more of a "technical detail API" of Kueue.

Comment (tenzen-y) [reply test/e2e/singlecluster/e2e_test.go:133]: It makes sense.

**Review by tenzen-y — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**